File: python_skeleton8/src/apriori.py
from typing import Optional
import numpy as np
import copy


def apriori(costs: np.ndarray, weights: Optional = None, order: Optional = None):
    """
    Implement the two example apriori methods presented in the lecture
    Parameters
    ----------
    costs   (n_points, m_costs) array
    weights (m_costs, ) array. Determines the weighting of the costs. If None use lexical ordering
    order   (m_costs, ) array. Determines the lexicograpical order. If None use weighted sum

    Returns
    -------
    Index of optimal element according to apriori method
    """
    if weights is not None and order is not None:
        raise Exception('You can only specify weight or order but not both')
    if weights:
        cost = np.dot(costs, weights)
        arg = np.argmin(cost)
    if order:
        order = np.array(order)
        costs_temp = copy.copy(costs)
        arg = np.arange(costs.shape[0])
        for ord in order:
            arg = np.argmin(costs_temp[:,ord])
            costs_temp[~np.isin(np.arange(costs_temp.shape[0]), arg),:] = np.Inf
        
    return arg

File: R_skeleton8/skeleton/apriori.R
# Implement the two example apriori methods presented in the lecture
# @param costs (`data.table()`): data.table of dimension (n_points, m_costs)
# @param weights (`double()`): vector of length (m_costs). Determines the weighting of the costs. If NULL use lexical ordering
# @param order (`double()`): vector of length (m_costs). Determines the lexicograpical order. If NULL use weighted sum
# @return (`integer(1)`): Index of optimal element according to apriori method
apriori = function (costs, weights = NULL, order = NULL) {
  mcosts = as.matrix(costs)
  if (!is.null(weights) && !is.null(order)) {
    stop("You can only specify weight or order but not both")
  }
  if (!is.null(weights)) {
    stop("To Implement")
  }
  if (!is.null(order)) {
    stop("To Implement")
  }
  arg
}
